mlir-sycl/lib/Transforms/Inliner.cpp

```diff
@@ -261,38 +547,59 @@ void Inliner::inlineCallsInSCC(Inliner &Inliner, CallGraphSCC &SCC) {
   // Try to inline each of the call operations. Don't cache the end iterator
   // here as more calls may be added during inlining.
   for (unsigned I = 0; I < Inliner.getCalls().size(); ++I) {
-    ResolvedCall It = Inliner.getCall(I);
-    CallOpInterface Call = It.Call;
-
-    bool DoInline = Inliner.shouldInline(It);
-    LLVM_DEBUG(llvm::dbgs() << ((DoInline) ? "* Inlining call: "
-                                           : "* Not inlining call: ")
-                            << I << ". " << Call << "\n";);
+    ResolvedCall ResolvedCall = Inliner.getCall(I);
+    bool DoInline = Inliner.shouldInline(ResolvedCall, UseList);
     if (!DoInline)
       continue;
 
-    Region *TgtRegion = It.TgtNode->getCallableRegion();
-    LogicalResult InlineRes = inlineCall(
-        Inliner, Call, cast<CallableOpInterface>(TgtRegion->getParentOp()),
-        TgtRegion, /*shouldCloneInlinedRegion=*/true);
+    LLVM_DEBUG(llvm::dbgs() << "* Inlining call: " << I << ". "
+                            << ResolvedCall.Call << "\n";);
+
+    Region *TgtRegion = ResolvedCall.TgtNode->getCallableRegion();
+    LogicalResult InlineRes =
+        inlineCall(Inliner, ResolvedCall.Call,
+                   cast<CallableOpInterface>(TgtRegion->getParentOp()),
+                   TgtRegion, /*shouldCloneInlinedRegion=*/true);
 
     if (failed(InlineRes)) {
       LLVM_DEBUG(llvm::dbgs() << "** Failed to inline\n");
       continue;
     }
-
     LLVM_DEBUG(llvm::dbgs() << "** Inline succeeded\n");
 
-    // then erase the call.
-    Call.erase();
+    NumInlinedCalls++;
```

Comment:
```suggestion
    ++NumInlinedCalls;
```

Reply:
In this context using pre-increment or post-increment doesn't make a difference.